packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -0,0 +1,326 @@
+        if (key.match(/^AWS_BATCH/)) {
+          throw new Error(
+            `Invalid environment variable name: ${key}. Environment variable names starting with 'AWS_BATCH' are reserved.`
+          );
+        }
+      });
+    }
+  }
+
+  public bind(_task: sfn.Task): sfn.StepFunctionsTaskConfig {
+    return {
+      resourceArn: getResourceArn(
+        'batch',
+        'submitJob',
+        this.integrationPattern
+      ),
+      policyStatements: [
+        // Resource-level access control is not supported by Batch
+        // https://docs.aws.amazon.com/step-functions/latest/dg/batch-iam.html
+        new iam.PolicyStatement({
+          resources: ['*'],
+          actions: ['batch:SubmitJob']
+        }),
+        new iam.PolicyStatement({
+          resources: [
+            Stack.of(_task).formatArn({
+              service: 'events',
+              resource: 'rule/StepFunctionsGetEventsForBatchJobsRule'
+            })
+          ],
+          actions: [
+            'events:PutTargets',
+            'events:PutRule',
+            'events:DescribeRule'
+          ]
+        })
+      ],
+      parameters: {
+        JobDefinition: this.props.jobDefinition.jobDefinitionArn,
+        JobName: this.props.jobName,
+        JobQueue: this.props.jobQueue.jobQueueArn,
+        Parameters: this.props.payload,
+
+        ArrayProperties:
+          this.props.arraySize !== undefined
+            ? { Size: this.props.arraySize }
+            : undefined,
+
+        ContainerOverrides: this.props.containerOverrides
+          ? this.configureContainerOverrides(this.props.containerOverrides)
+          : undefined,
+
+        DependsOn: this.props.dependsOn
+          ? this.props.dependsOn.map(jobDependency => ({
+              JobId: jobDependency.jobId,
+              Type: jobDependency.type
+            }))
+          : undefined,
+
+        RetryStrategy:
+          this.props.attempts !== undefined
+            ? { Attempts: this.props.attempts }
+            : undefined,
+
+        Timeout: this.props.timeout
+          ? { AttemptDurationSeconds: this.props.timeout.toSeconds() }
+          : undefined
+      }
+    };
+  }
+
+  private configureContainerOverrides(containerOverrides: ContainerOverrides) {
+    return {
+      Command: containerOverrides.command,
+      Environment: containerOverrides.environment
+        ? Object.entries(containerOverrides.environment).map(
+            ([key, value]) => ({
+              Name: key,
+              Value: value
+            })
+          )
+        : undefined,
+      InstanceType: containerOverrides.instanceType?.toString(),
+      Memory: containerOverrides.memory,
+      ResourceRequirements: containerOverrides.gpuCount
+        ? [
+            {
+              Type: 'GPU',
+              Value: `${containerOverrides.gpuCount}`
+            }
+          ]
+        : undefined,
+      Vcpus: containerOverrides.vcpus
```

Comment:
Let's make this more imperative. The same unreadability from the previous iteration exists here as well.

```ts
const environment = Object.entries(containerOverrides.environment ?? {}).map(...);

let resources = undefined;
if (containerOverrides.gpuCound) {
  resources = [ { ... } ];
}

...
...

return {
  Envrionment: containerOverrides.environment ? environment : undefined,
  ResourceRequirements: resources,
  ...
};
```